File: backend/src/CloudCrafter.Domain/Domain/Teams/SimpleTeamWithProjectsAndEnvironmentsDto.cs
using AutoMapper;
using CloudCrafter.Domain.Domain.Project;
using CloudCrafter.Domain.Entities;

namespace CloudCrafter.Domain.Domain.Teams;

public class SimpleTeamWithProjectsAndEnvironmentsDto
{
    public required Guid Id { get; init; }
    public required string Name { get; set; }

    public List<ProjectDto> Projects { get; init; } = new();

    private class Mapping : Profile
    {
        public Mapping()
        {
            CreateMap<Team, SimpleTeamWithProjectsAndEnvironmentsDto>();
        }
    }
}

File: frontend/packages/api/src/__generated__/types/SimpleTeamWithProjectsAndEnvironmentsDto.ts
import type { ProjectDto } from './ProjectDto'

export type SimpleTeamWithProjectsAndEnvironmentsDto = {
  /**
   * @type string, uuid
   */
  id: string
  /**
   * @type string
   */
  name: string
  /**
   * @type array
   */
  projects: ProjectDto[]
}